File: Formularios/app/src/main/java/com/oscarmena/formularios/consultar.java
package com.oscarmena.formularios;

import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import android.widget.Toast;

import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ListView;
import java.util.ArrayList;
import java.util.List;

public class consultar extends AppCompatActivity {

    private EditText etDocumento, etNombreCatedra;
    private Button btnConsultaIdEstudiante, btnConsultaIdCatedra, btnVolver;
    private ListView lvEstudiante, lvCatedras;
    private DBHandler dbHandler;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_consultar);

        // Inicializar vistas
        etDocumento = findViewById(R.id.etDocumento);
        etNombreCatedra = findViewById(R.id.etNombreCatedra);
        btnConsultaIdEstudiante = findViewById(R.id.btnConsultaIdEstudiante);
        btnConsultaIdCatedra = findViewById(R.id.btnConsultaIdCatedra);
        btnVolver = findViewById(R.id.btnVolver);
        lvEstudiante = findViewById(R.id.lvEstudiante);
        lvCatedras = findViewById(R.id.lvCatedras);

        dbHandler = new DBHandler(this);

        // Acción para consultar estudiante por documento
        btnConsultaIdEstudiante.setOnClickListener(v -> consultarEstudiante());

        // Acción para consultar cátedra por nombre
        btnConsultaIdCatedra.setOnClickListener(v -> consultarCatedra());

        // Volver a la pantalla anterior
        btnVolver.setOnClickListener(v -> finish());
    }

    // Método para consultar estudiante
    private void consultarEstudiante() {
        String documento = etDocumento.getText().toString().trim();
        if (!documento.isEmpty()) {
            List<String> datosEstudiante = dbHandler.consultarEstudiante(documento);
            if (!datosEstudiante.isEmpty()) {
                ArrayAdapter<String> adapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, datosEstudiante);
                lvEstudiante.setAdapter(adapter);
            } else {
                Toast.makeText(this, "No se encontraron estudiantes con ese documento", Toast.LENGTH_SHORT).show();
            }
        } else {
            Toast.makeText(this, "Ingrese un documento", Toast.LENGTH_SHORT).show();
        }
    }

    // Método para consultar cátedra
    private void consultarCatedra() {
        String nombre = etNombreCatedra.getText().toString().trim();
        String documento = etDocumento.getText().toString().trim(); // Asegurar que se tiene el documento

        if (!nombre.isEmpty() && !documento.isEmpty()) { // Verifica ambos valores
            List<String> datosCatedra = dbHandler.consultarCatedra(nombre, documento);

            if (!datosCatedra.isEmpty()) {
                ArrayAdapter<String> adapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, datosCatedra);
                lvCatedras.setAdapter(adapter);
            } else {
                Toast.makeText(this, "No se encontraron cátedras con ese nombre para este estudiante", Toast.LENGTH_SHORT).show();
            }
        } else {
            Toast.makeText(this, "Ingrese el nombre de la cátedra y el documento del estudiante", Toast.LENGTH_SHORT).show();
        }
    }
}

File: FormulariosKotlin/app/src/main/java/com/oscarmena/formularioskotlin/Consultar.kt
package com.oscarmena.formularioskotlin

import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.EditText
import android.widget.ListView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class Consultar : AppCompatActivity() {
    private lateinit var etDocumento: EditText
    private lateinit var etNombreCatedra: EditText
    private lateinit var btnConsultaIdEstudiante: Button
    private lateinit var btnConsultaIdCatedra: Button
    private lateinit var btnVolver: Button
    private lateinit var lvEstudiante: ListView
    private lateinit var lvCatedras: ListView
    private lateinit var dbHandler: DBHandler

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_consultar)

        // Initialize views
        etDocumento = findViewById(R.id.etDocumento)
        etNombreCatedra = findViewById(R.id.etNombreCatedra)
        btnConsultaIdEstudiante = findViewById(R.id.btnConsultaIdEstudiante)
        btnConsultaIdCatedra = findViewById(R.id.btnConsultaIdCatedra)
        btnVolver = findViewById(R.id.btnVolver)
        lvEstudiante = findViewById(R.id.lvEstudiante)
        lvCatedras = findViewById(R.id.lvCatedras)

        dbHandler = DBHandler(this)

        // Action to query student by document
        btnConsultaIdEstudiante.setOnClickListener { consultarEstudiante() }

        // Action to query class by name
        btnConsultaIdCatedra.setOnClickListener { consultarCatedra() }

        // Return to previous screen
        btnVolver.setOnClickListener { finish() }
    }

    // Method to query student
    private fun consultarEstudiante() {
        val documento = etDocumento.text.toString().trim()
        if (documento.isNotEmpty()) {
            val datosEstudiante = dbHandler.consultarEstudiante(documento)
            if (datosEstudiante.isNotEmpty()) {
                val adapter = ArrayAdapter(
                    this,
                    android.R.layout.simple_list_item_1,
                    datosEstudiante
                )
                lvEstudiante.adapter = adapter
            } else {
                Toast.makeText(
                    this,
                    "No se encontraron estudiantes con ese documento",
                    Toast.LENGTH_SHORT
                ).show()
            }
        } else {
            Toast.makeText(
                this,
                "Ingrese un documento",
                Toast.LENGTH_SHORT
            ).show()
        }
    }

    // Method to query class
    private fun consultarCatedra() {
        val nombre = etNombreCatedra.text.toString().trim()
        val documento = etDocumento.text.toString().trim()

        // Ensure both values are present
        if (nombre.isNotEmpty() && documento.isNotEmpty()) {
            val datosCatedra = dbHandler.consultarCatedra(nombre, documento)
            if (datosCatedra.isNotEmpty()) {
                val adapter = ArrayAdapter(
                    this,
                    android.R.layout.simple_list_item_1,
                    datosCatedra
                )
                lvCatedras.adapter = adapter
            } else {
                Toast.makeText(
                    this,
                    "No se encontraron cátedras con ese nombre para este estudiante",
                    Toast.LENGTH_SHORT
                ).show()
            }
        } else {
            Toast.makeText(
                this,
                "Ingrese el nombre de la cátedra y el documento del estudiante",
                Toast.LENGTH_SHORT
            ).show()
        }
    }
}